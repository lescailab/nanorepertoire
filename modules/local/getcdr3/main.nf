process GETCDR3 {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/biopython:1.78':
        'ghcr.io/nibscbioinformatics/biopython:v1.78' }"

    input:
    tuple val(meta), path(translated)


    output:
    tuple val("${meta.id}"), val("${meta.immunisation}"), val("${meta.boost}"), path("*.fasta"), emit: fasta
    tuple val(meta), path("*.hist")                                                            , emit: hist
    tuple val(meta), path("*.tsv")                                                             , emit: tsv
    path '*.hist'                                                                              , emit: histonly
    path '*.tsv'                                                                               , emit: tsvonly
    val meta                                                                                   , emit: metaonly
    path "versions.yml"                                                                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    getcdr3.py \
    -i ${translated} \
    -c ${meta.id}_cdr3.fasta \
    -o ${meta.id}_cdr3.hist \
    -t ${meta.id}_cdr3.tsv \
    -s ${meta.id} \
    -m ${meta.immunisation} \
    -b ${meta.boost}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """

}
