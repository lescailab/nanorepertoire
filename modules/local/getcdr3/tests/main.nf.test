nextflow_process {

    name "Test Process GETCDR3"
    script "../main.nf"
    process "GETCDR3"
    config "./nextflow.config"
    tag "nanobodies"
    tag "modules"
    tag "modules_nfcore"
    tag "getcdr3"

    test("getcdr3-test") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test1', single_end:false ], // meta map
                    [file("https://github.com/lescailab/testdata/raw/master/nanobodies/test01_aa.clusters", checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
