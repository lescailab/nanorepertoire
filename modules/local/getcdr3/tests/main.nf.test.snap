{
    "getcdr3-test": {
        "content": [
            {
                "0": [
                    [
                        "test1",
                        "null",
                        "null",
                        "test1_cdr3.fasta:md5,14bada664572f90491234c861e82de5c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_cdr3.hist:md5,b5ed97b4d9062117e8b43edde36ddb74"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_cdr3.tsv:md5,8cf072abda52c6d11b6f0522d54f0e38"
                    ]
                ],
                "3": [
                    "test1_cdr3.hist:md5,b5ed97b4d9062117e8b43edde36ddb74"
                ],
                "4": [
                    "test1_cdr3.tsv:md5,8cf072abda52c6d11b6f0522d54f0e38"
                ],
                "5": [
                    {
                        "id": "test1",
                        "single_end": false
                    }
                ],
                "6": [
                    "versions.yml:md5,a11ef45f3e75cbd545cc0ec3d3d17b82"
                ],
                "fasta": [
                    [
                        "test1",
                        "null",
                        "null",
                        "test1_cdr3.fasta:md5,14bada664572f90491234c861e82de5c"
                    ]
                ],
                "hist": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_cdr3.hist:md5,b5ed97b4d9062117e8b43edde36ddb74"
                    ]
                ],
                "histonly": [
                    "test1_cdr3.hist:md5,b5ed97b4d9062117e8b43edde36ddb74"
                ],
                "metaonly": [
                    {
                        "id": "test1",
                        "single_end": false
                    }
                ],
                "tsv": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_cdr3.tsv:md5,8cf072abda52c6d11b6f0522d54f0e38"
                    ]
                ],
                "tsvonly": [
                    "test1_cdr3.tsv:md5,8cf072abda52c6d11b6f0522d54f0e38"
                ],
                "versions": [
                    "versions.yml:md5,a11ef45f3e75cbd545cc0ec3d3d17b82"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T15:35:22.411639"
    }
}